test/test_jit.py

```diff
@@ -2210,9 +2210,158 @@ def forward(self, v):
                 for m in self.mods:
                     print(m)
                 return v
-        with self.assertRaisesRegex(RuntimeError, "is not iterable"):
+        with self.assertRaisesRegex(RuntimeError, "cannot be used as tuple"):
             M()
 
+    def test_script_star_expr(self):
+        class M(torch.nn.Module):
+            def __init__(self):
+                super(M, self).__init__()
+
+            def forward(self, *inputs):
+                output = inputs[0]
+                for i in range(1, len(inputs)):
+                    output += inputs[i]
+                return output
+
+        class Mult(torch.nn.Module):
+            def __init__(self):
+                super(Mult, self).__init__()
+
+            def forward(self, rep):
+                return rep, rep, rep
+
+        class M2(torch.jit.ScriptModule):
+            def __init__(self):
+                super(M2, self).__init__(True)
+                self.m = torch.jit.trace(
+                    torch.ones(4, 3), torch.ones(4, 3), torch.ones(4, 3))(M())
+                self.g = torch.jit.trace(torch.ones(4, 3))(Mult())
+
+            @torch.jit.script_method
+            def forward(self, rep):
+                tup = self.g(rep)
+                return self.m(*tup)
+
+        m = M2()
+        self.assertEqual(m(torch.zeros(4, 3)), 3 * torch.zeros(4, 3))
+
+    def test_script_star_expr_string(self):
+        class M(torch.nn.Module):
+            def __init__(self):
+                super(M, self).__init__()
+
+            def forward(self, *inputs):
+                output = inputs[0]
+                for i in range(1, len(inputs)):
+                    output += inputs[i]
+                return output
+
+        class Mult(torch.nn.Module):
+            def __init__(self):
+                super(Mult, self).__init__()
+
+            def forward(self, rep):
+                return rep, rep, rep
+
+        class M2(torch.jit.ScriptModule):
+            def __init__(self):
+                super(M2, self).__init__(True)
+                self.m = torch.jit.trace(
+                    torch.ones(4, 3), torch.ones(4, 3), torch.ones(4, 3))(M())
+                self.g = torch.jit.trace(torch.ones(4, 3))(Mult())
+
+                self.define('''
+            def forward(self, rep):
+                tup = self.g(rep)
+                return self.m(*tup)
+                ''')
+
+        m = M2()
+        self.assertEqual(m(torch.zeros(4, 3)), 3 * torch.zeros(4, 3))
+
+    def test_script_star_assign(self):
+        class M(torch.nn.Module):
```

Comment:
This module is repeated over and over. Can we extract it and share between the tests?